vault_core/src/vault.rs

```diff
@@ -794,9 +797,23 @@ impl Vault {
             return Err(VaultError::VaultInsufficientFunds);
         }
 
-        let fee_amount = self.calculate_withdraw_fee(amount_in)?;
+        let program_fee_amount = Config::calculate_program_fee(program_fee_bps, amount_in)?;
```

Comment:
what're thoughts on calculating the vault fee as a % of amount_in - program_fee_amount? i think makes the code slightly simpler? but some tradeoffs. lmk wyt

Reply:
what would calculate_vault_fee take in and return in this case?